feat(marks): marks keep "view" info and restore it on jump (#15831)
neovim/neovim: runtime/doc/options.txt

			jumplist and then jumping to a location.
			|jumplist-stack|

> 	  view          When moving on the jumplist try to recover the view in
> 			which the jump was defined. The view of a jump is

Reviewer: ```suggestion
	  view          When moving through the jumplist try to restore the view from
			when the jump occurred. The view of a jump is
```

			jumplist and then jumping to a location.
			|jumplist-stack|

	  view          When moving through the jumplist, changelist, and
			|alternate-file| try to restore the |mark-view| in

> 			which the jump occurred.

Reviewer 2: Why not also make it apply to `'x` ? Then we can drop the `z'` / `[count]'` variants, at least until we get strong feedback showing a use-case for _not_ restoring the view.
